File: app/src/main/java/com/xmartlabs/template/controller/SessionController.java
package com.xmartlabs.template.controller;

import android.support.annotation.NonNull;

import com.annimon.stream.Optional;
import com.xmartlabs.bigbang.core.controller.CoreSessionController;
import com.xmartlabs.bigbang.core.model.SessionType;
import com.xmartlabs.template.model.Session;

public class SessionController extends CoreSessionController {
  public SessionController(Class<? extends SessionType> clazz) {
    super(clazz);
  }

  @NonNull
  @Override
  public Optional<Session> getSession() {
    //noinspection unchecked
    return (Optional<Session>) super.getSession();
  }
}

File: app/src/main/java/com/xmartlabs/template/controller/SessionController.kt
package com.xmartlabs.template.controller

import com.xmartlabs.bigbang.core.controller.CoreSessionController
import com.xmartlabs.bigbang.core.controller.SharedPreferencesController
import com.xmartlabs.template.model.Session
import javax.inject.Inject

class SessionController @Inject constructor(sharedPreferencesController: SharedPreferencesController)
  : CoreSessionController(sharedPreferencesController) {
  override fun getSessionType() = Session::class.java

  val session
    get() = abstractSession as? Session?
}
